modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java

```diff
@@ -0,0 +1,176 @@
+package org.testcontainers.containers;
+
+import static org.junit.Assert.assertEquals;
+
+import com.influxdb.client.InfluxDBClient;
+import com.influxdb.client.InfluxDBClientFactory;
+import com.influxdb.client.InfluxDBClientOptions;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import lombok.SneakyThrows;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+/**
+ * See
+ * <a href="https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0">https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0</a>
+ */
+public class InfluxDBContainerV2<SELF extends InfluxDBContainerV2<SELF>> extends GenericContainer<SELF> {
+
+    public static final Integer INFLUXDB_PORT = 8086;
+
+    private static final String REGISTRY = "quay.io";
+    private static final String REPOSITORY = "influxdb/influxdb";
+    private static final String TAG = "v2.0.0";
+    private static final DockerImageName DEFAULT_IMAGE_NAME =
+        DockerImageName.parse(String.format("%s/%s:%s", REGISTRY, REPOSITORY, TAG));
+    private static final int NO_CONTENT_STATUS_CODE = 204;
+    private static final String INFLUX_SETUP_SH = "influx-setup.sh";
+
+    private String username = "test-user";
+    private String password = "test-password";
+    private String bucket = "test-bucket";
+    private String organization = "test-org";
+    private int retention = 0;
+    private String retentionUnit = RetentionUnit.NANOSECONDS.label;
+
+    private InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
```

Comment:
Running with

```java
InfluxDBContainerV2.createWithSpecificTag(DockerImageName.parse("quay.io/influxdb/influxdb:v2.0.7"));
```

will fail with:

```
aused by: java.lang.IllegalStateException: Failed to verify that image 'quay.io/influxdb/influxdb:v2.0.7' is a compatible substitute for 'quay.io/influxdb/influxdb:v2.0.0'. This generally means that you are trying to use an image that Testcontainers has not been designed to use. If this is deliberate, and if you are confident that the image is compatible, you should declare compatibility in code using the `asCompatibleSubstituteFor` method. For example:
   DockerImageName myImage = DockerImageName.parse("quay.io/influxdb/influxdb:v2.0.7").asCompatibleSubstituteFor("quay.io/influxdb/influxdb:v2.0.0");
and then use `myImage` instead.
	at org.testcontainers.utility.DockerImageName.assertCompatibleWith(DockerImageName.java:248)
	at org.testcontainers.containers.InfluxDBContainerV2.<init>(InfluxDBContainerV2.java:42)
	at org.testcontainers.containers.InfluxDBContainerV2.createWithSpecificTag(InfluxDBContainerV2.java:57)
	at com.kiwigrid.ence.timeseriesservice.config.TestInfluxDbConfig.influxDBContainer(TestInfluxDbConfig.java:21)
```

Reply:
Good spot! `assertCompatibleWith` will look for an exact match on the tag part, if it's set. Usually we should have a `DockerImageName` that has no tag set passed in as a parameter.

If we go with my comment just below about constructors, we'd expect there to be zero occurrences of a tag string in this class.